### ISSUE TITLE
How to make dependencies available in tests

### ISSUE THREAD

**Issue by jennybc** [hadley/r-pkgs#580]
General observation, sharpened by working on tidyr revdeps: would be good to clarify how one should call functions in other packages in code below `tests/`. Address this for packages that appear in Imports vs Suggests. Address for code re: helpers & setup vs. inside a test.

**Comment by jennybc** [comment]
For example, in a regular test file, should we ever see a `library()` call? I think not, but am also not 100% sure about this.

**Comment by jimhester** [comment]
I don't think you ever need `library()`. If it is in `Imports` you shouldn't need the `library()` at all, if in `Suggests` is should really be a `if (require())` or `if (requireNamespace()) ` so the test doesn't fail if the Suggested package is not installed.

**Comment by jyuu** [comment]
Would love to know what's the best practice.. for `butcher` we don't want to include too many dependencies under `Imports` so a lot of the unit tests rely on the `library()` call. 

https://github.com/tidymodels/butcher/tree/master/tests/testthat

**Comment by hadley** [comment]
@jimhester I thought we assumed that "Suggests" effectively means development dependency, and that generally all suggested packages are installed when testing (the only exception is packages that are known to be hard to install or missing on some platforms)

**Comment by hadley** [comment]
@jyuu in https://github.com/tidymodels/butcher/blob/master/tests/testthat/test-kknn.R (e.g.), I think you could replace `library(kknn)` with with `kknn::kknn()` — I'd imagine in most cases you'd only need a couple of qualified calls to the package.
